tests/app/main/test_briefs.py

```diff
@@ -1912,6 +1912,44 @@ def test_supplier_details_shown(self, data_api_client):
         assert requirements_data.row(2).cell(0) == "Email address"
         assert requirements_data.row(2).cell(1) == "contact@big.com"
 
+    def test_analytics_and_messages_applied_on_first_submission(self, data_api_client):
+        """Go through submitting to edit_brief_response and the redirect to view_response_result. Assert messages."""
```

Comment:
It looks like our other tests that test flash messages on this page are more unit (checking the flash is set on the previous page and then asserting if you visit this page with a flash message set that it shows the correct message). Either way works with me but just thought I'd point it out. Would be nice if our tests were more consistent but I don't think you should go to the effort of refactoring tests for that reason